### PR TITLE
Feature/ASSA-10 Additional detekt configurations for Compose and code cleanup

### PR DESCRIPTION
**Pull request by Josh-Gebbeken**
Added additional Detekt configuration for Compose:

- Magic Numbers for Compose like 32.dp 
- PreviewAllDevices annotation
- File-wide suppression for ther Dimens class like ConstructorParameter, MatchingDeclarationName, LongParameterList
- Additional work to break down long methods
- Refactored files that don't have a matching class name to match. Refactored based on actual name of screen.
- Fixed additional UnusedPrivateMembers

### CODE REVIEW

File: app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/Routes.kt [hunk 1]
```diff
@@ -12,12 +12,12 @@ object Routes {
     const val Snippets = "snippets"
     const val Profile = "profile"
 
-    fun RepositoryBrowser(data: RepositoryBrowserData) =
+    fun repositoryBrowser(data: RepositoryBrowserData) =
```
Comment (jaredstockton) [inline app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/Routes.kt]: @Josh-Gebbeken @BottleRocket-Colin I'm honestly on the fence about this one. If we had used sealed classes (which we have already discussed and decided against as they don't bring a lot to the table) then all of the sub objects and data classes would be UpperCamelCased.

Since we're using an Object, detekt is going to complain when we use UpperCamelCase function names that take in additional route data to mimic the casing for sealed classes/other Composable functions.

Thoughts?

File: compose/src/main/java/com/bottlerocketstudios/compose/devoptions/DevOptionsScreen.kt [hunk 2]
```diff
@@ -48,38 +48,13 @@ import com.bottlerocketstudios.launchpad.compose.bold
...
-@SuppressLint("UnusedMaterialScaffoldPaddingParameter")
 @Composable
 fun DevOptionsScreen(state: DevOptionsState) {
     DevOptionsScreenTheme {
-        Scaffold(floatingActionButton = { FabLayout(state.onRestartCtaClick) }) {
+        Scaffold(floatingActionButton = { FabLayout(state.onRestartCtaClick) }) { paddingValues ->
             Column(
                 modifier = Modifier
-                    .fillMaxSize()
+                    .fillMaxSize().padding(paddingValues)
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/devoptions/DevOptionsScreen.kt:56]: @Josh-Gebbeken Was this addition intentional?

File: compose/src/main/java/com/bottlerocketstudios/compose/repository/FileBrowser.kt [hunk 3]
```diff
@@ -27,11 +27,6 @@ import com.bottlerocketstudios.compose.util.convertToImageBitmap
 import com.bottlerocketstudios.launchpad.compose.light
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/repository/FileBrowser.kt:25]: @Josh-Gebbeken Should this file be refactored to `FileBrowserScreen`?

File: compose/src/main/java/com/bottlerocketstudios/compose/repository/RepositoryBrowserScreen.kt [hunk 4]
```diff
@@ -188,3 +180,12 @@ private fun PreviewOuterScreenContent() {
...
 }
+
+data class RepositoryBrowserScreenState(
+    val path: State<String>,
+    val itemCount: State<Int>,
+    val repositoryItems: State<List<RepositoryItemUiModel>>,
+    val onRepositoryItemClicked: (RepositoryItemUiModel) -> Unit
+)
+
+private const val DEFAULT_ITEM_COUNT = 3
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/repository/RepositoryBrowserScreen.kt]: @Josh-Gebbeken This provides more visibility when pulled out 👍🏻

File: compose/src/main/java/com/bottlerocketstudios/compose/resources/Theme.kt [hunk 5]
```diff
@@ -19,6 +19,7 @@ fun ProvideColors(
 private val LocalAppColors = staticCompositionLocalOf {
     lightColors
 }
+private const val SMALL_SCREEN_WIDTH = 360
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/resources/Theme.kt]: @Josh-Gebbeken nit: Refactor name to `SMALL_SCREEN_WIDTH_DP` for clarity on the unit of the value. If it stays as is, you would need to reference its usage to determine if the value represented dps or pixels :)

File: domain/src/main/java/com/bottlerocketstudios/brarchitecture/domain/models/Status.kt [hunk 6]
```diff
@@ -96,4 +96,5 @@ suspend fun <T : Any> logWrappedExceptions(className: String, methodName: String
 
 // Simple version of wrap exceptions; with no logging.
 suspend fun <T : Any> wrapExceptions(block: suspend () -> Status<T>): Status<T> =
+    @Suppress("TooGenericExceptionCaught") // this might be overly broad (catching NPEs isn't ideal) but not worth it yet to check for all other possible types (IO..., Json..., others)
```
Comment (jaredstockton) [inline domain/src/main/java/com/bottlerocketstudios/brarchitecture/domain/models/Status.kt]: @Josh-Gebbeken nit: Refactor comment to use `catching general Exceptions isn't ideal` instead of `catching NPEs isn't ideal`

File: config/detekt/detekt.yml [hunk 7]
```diff
@@ -681,7 +680,7 @@ style:
       - '1'
       - '2'
     ignoreHashCodeFunction: true
-    ignorePropertyDeclaration: false
+    ignorePropertyDeclaration: true
```
Comment (jaredstockton) [inline config/detekt/detekt.yml:683]: @Josh-Gebbeken I'm just curious where you see instances in the codebase where this rule is broken that caused you to change the value to true. Can you share a few?

**Review by jaredstockton — CHANGES_REQUESTED** [review]
@Josh-Gebbeken Thanks for the PR! I've left some comments of things to address prior to approval :)

File: compose/src/main/java/com/bottlerocketstudios/compose/appbar/ArchAppBar.kt [hunk 8]
```diff
@@ -62,3 +58,9 @@ fun ArchAppBar(
         )
     }
 }
+
```
Comment (BottleRocket-Colin) [inline compose/src/main/java/com/bottlerocketstudios/compose/appbar/ArchAppBar.kt]: Curious as to why this was moved below the composable function.  This makes it slightly harder to find.  Potentially much harder, if the file has a lot of composable functions after the main one.

File: compose/src/main/java/com/bottlerocketstudios/compose/devoptions/DevOptionsScreen.kt (hunk 2 above)
Comment (BottleRocket-Colin) [reply compose/src/main/java/com/bottlerocketstudios/compose/devoptions/DevOptionsScreen.kt:56]: @br-jared-stockton - This is following the convention defined here: https://developer.android.com/jetpack/compose/layouts/material#scaffold

Good enhancement @Josh-Gebbeken 

I was being sloppy by adding the suppress notation.

File: compose/src/main/java/com/bottlerocketstudios/compose/home/HomeCard.kt [hunk 9]
```diff
@@ -66,43 +66,8 @@ fun CardLayout(userRepositoryUiModel: UserRepositoryUiModel, selectItem: (userRe
...
-                        modifier = Modifier
-                            .padding(
-                                top = Dimens.grid_0_5,
-                                bottom = Dimens.grid_1
-                            )
-                            .wrapContentHeight()
-                            .fillMaxWidth()
-                    )
-                }
+               RepoDescriptionComponent(userRepositoryUiModel = userRepositoryUiModel, Modifier.weight(2f) )
```
Comment (BottleRocket-Colin) [inline compose/src/main/java/com/bottlerocketstudios/compose/home/HomeCard.kt]: In an effort to make this a bit more generic, lets change function signature of RepoDescriptionContent to accept each value: repo.name, display name, and description.  That way it's not coupled to UI model.

File: compose/src/main/java/com/bottlerocketstudios/compose/profile/ProfileScreen.kt [hunk 10]
```diff
@@ -31,50 +30,14 @@ import com.bottlerocketstudios.compose.util.PreviewAllDevices
...
-            state.nickname.value,
-            color = ArchitectureDemoTheme.colors.onSurface,
-            style = MaterialTheme.typography.h4.normal(),
-            modifier = Modifier
-                .padding(
-                    top = Dimens.grid_0_5
-                )
-                .wrapContentSize()
-        )
+        ProfileDetailComponent(state = state)
```
Comment (BottleRocket-Colin) [inline compose/src/main/java/com/bottlerocketstudios/compose/profile/ProfileScreen.kt]: Same as an earlier comment, let's pull out just what is needed from state and pass that along to ProfileDetailComponent.

File: compose/src/main/java/com/bottlerocketstudios/compose/repository/FileBrowser.kt (hunk 3 above)
Comment (BottleRocket-Colin) [reply compose/src/main/java/com/bottlerocketstudios/compose/repository/FileBrowser.kt:25]: Couldn't hurt.

File: compose/src/main/java/com/bottlerocketstudios/compose/repository/RepositoryBrowserScreen.kt (hunk 4 above)
Comment (BottleRocket-Colin) [reply compose/src/main/java/com/bottlerocketstudios/compose/repository/RepositoryBrowserScreen.kt]: @Josh-Gebbeken - Since this value needs to stay in sync with the list<RepositoryItemUiModel> it may be better to val that list a constant and then get the size of the list.  This would avoid maintaining this constant.

File: config/detekt/detekt.yml (hunk 7 above)
Comment (BottleRocket-Colin) [reply config/detekt/detekt.yml:683]: @br-jared-stockton - I think this is for the colors file ?  Is that right @Josh-Gebbeken ?

**Review by BottleRocket-Colin — CHANGES_REQUESTED** [review]
A couple of minor nits in my comments. Only major issue I have is the moving of State objects below the screen in compose module.  Curious as to what error Detekt was throwing and if it could be suppressed for that file.  I did that b/c there's a lot of back and forth scrolling and it felt faster to have the state at the top of file near the top of the screen composable.

File: compose/src/main/java/com/bottlerocketstudios/compose/devoptions/DevOptionsScreen.kt (hunk 2 above)
Comment (Josh-Gebbeken) [reply compose/src/main/java/com/bottlerocketstudios/compose/devoptions/DevOptionsScreen.kt:56]: It was highlighting in red for me so I ended up adding it to remove.

File: compose/src/main/java/com/bottlerocketstudios/compose/home/HomeCard.kt (hunk 9 above)
Comment (Josh-Gebbeken) [reply compose/src/main/java/com/bottlerocketstudios/compose/home/HomeCard.kt]: Sounds like a good idea.

File: compose/src/main/java/com/bottlerocketstudios/compose/profile/ProfileScreen.kt (hunk 10 above)
Comment (Josh-Gebbeken) [reply compose/src/main/java/com/bottlerocketstudios/compose/profile/ProfileScreen.kt]: ok

File: compose/src/main/java/com/bottlerocketstudios/compose/repository/FileBrowser.kt (hunk 3 above)
Comment (Josh-Gebbeken) [reply compose/src/main/java/com/bottlerocketstudios/compose/repository/FileBrowser.kt:25]: yeah that should work

File: compose/src/main/java/com/bottlerocketstudios/compose/resources/Theme.kt (hunk 5 above)
Comment (Josh-Gebbeken) [reply compose/src/main/java/com/bottlerocketstudios/compose/resources/Theme.kt]: ok

File: config/detekt/detekt.yml (hunk 7 above)
Comment (Josh-Gebbeken) [reply config/detekt/detekt.yml:683]: Yes for the magic numbers in the compose color for example val black = Color(0xff000000)

File: domain/src/main/java/com/bottlerocketstudios/brarchitecture/domain/models/Status.kt (hunk 6 above)
Comment (Josh-Gebbeken) [reply domain/src/main/java/com/bottlerocketstudios/brarchitecture/domain/models/Status.kt]: That was a copy pasta of another comment on this file. I can change that as well as this for better clarification.

File: compose/src/main/java/com/bottlerocketstudios/compose/appbar/ArchAppBar.kt (hunk 8 above)
Comment (Josh-Gebbeken) [reply compose/src/main/java/com/bottlerocketstudios/compose/appbar/ArchAppBar.kt]: This was one of the MatchingDeclarationName issues listed on Detekt. I thought It might of been easier to move it to the bottom and so this function can satisfy the issue.

Comment (Josh-Gebbeken) [reply compose/src/main/java/com/bottlerocketstudios/compose/appbar/ArchAppBar.kt]: Discussed over standup. We will move the Different Data Class states and put it in its own file in the same package

File: compose/src/main/java/com/bottlerocketstudios/compose/profile/ProfileScreen.kt (hunk 10 above)
Comment (Josh-Gebbeken) [reply compose/src/main/java/com/bottlerocketstudios/compose/profile/ProfileScreen.kt]: Moved to its own file within the same package

File: compose/src/main/java/com/bottlerocketstudios/compose/repository/FileBrowser.kt (hunk 3 above)
Comment (Josh-Gebbeken) [reply compose/src/main/java/com/bottlerocketstudios/compose/repository/FileBrowser.kt:25]: This has changed to `FileBrowserScreen`

File: compose/src/main/java/com/bottlerocketstudios/compose/home/HomeCard.kt (hunk 9 above)
Comment (Josh-Gebbeken) [reply compose/src/main/java/com/bottlerocketstudios/compose/home/HomeCard.kt]: Refactored out UI model so the component is decoupled.

File: config/detekt/detekt.yml (hunk 7 above)
Comment (jaredstockton) [reply config/detekt/detekt.yml:683]: Looks like this is one of the compose related solutions mentioned in the detekt docs: https://detekt.dev/docs/introduction/compose/#magicnumber

This should be fine to keep as is :)

File: app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/Routes.kt (hunk 1 above)
Comment (jaredstockton) [reply app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/Routes.kt]: @BottleRocket-Colin Any input on this one?

Comment (BottleRocket-Colin) [reply app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/Routes.kt]: Since it's a noun and it maintains consistency with the rest of  file, I'd say we should add a suppress notation to Routes for this one and keep it PascalCase.

Comment (Josh-Gebbeken) [reply app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/Routes.kt]: Changed back to PascalCase and added the suppression

**Comment by jaredstockton** [comment]
> A couple of minor nits in my comments. Only major issue I have is the moving of State objects below the screen in compose module. Curious as to what error Detekt was throwing and if it could be suppressed for that file. I did that b/c there's a lot of back and forth scrolling and it felt faster to have the state at the top of file near the top of the screen composable.

@Josh-Gebbeken Will detekt complain if we keep the State objects at the top of the composable screen files? I agree with @BottleRocket-Colin that it would be nicer to see the contract right at the top and then have the actual screen composable implementation below :)

**Comment by Josh-Gebbeken** [comment]
> > A couple of minor nits in my comments. Only major issue I have is the moving of State objects below the screen in compose module. Curious as to what error Detekt was throwing and if it could be suppressed for that file. I did that b/c there's a lot of back and forth scrolling and it felt faster to have the state at the top of file near the top of the screen composable.
> 
> @Josh-Gebbeken Will detekt complain if we keep the State objects at the top of the composable screen files? I agree with @BottleRocket-Colin that it would be nicer to see the contract right at the top and then have the actual screen composable implementation below :)

Detekt will complain it being on top. Colin and I ended up agreeing to just put them in a separate file in the same package

**Review by jaredstockton — APPROVED** [review]

**Review by BottleRocket-Colin — APPROVED** [review]
